### PR TITLE
CFE-3558: Added tail & tail_n to standard library

### PR DESCRIPTION
**Pull request by larsewi**

### CODE REVIEW

File: lib/reports.cf [hunk 1]
```diff
@@ -16,10 +16,28 @@ body printfile head(file)
 }
 
 body printfile head_n(file, n)
-# @brief Report the first 10 lines of a file
+# @brief Report the first n lines of a file
```
Comment (nickanderson) [inline lib/reports.cf:19]: Great, I would put backticks around n (`n`)

Comment (larsewi) [reply lib/reports.cf:19]: Yes, I agree

File: lib/reports.cf [hunk 2]
```diff
@@ -16,10 +16,28 @@ body printfile head(file)
...
         number_of_lines => "$(n)";
 }
+
+body printfile tail(file)
+# @brief Report the last 10 lines of a file
+# @param file The full path of the file to report
+{
+        file_to_print => "$(file)";
+      # GNU head defaults to 10, but we start counting from 0
+        number_of_lines => "-10";
```
[the remark this answers is not in the document]
Comment (larsewi) [reply lib/reports.cf:33]: The last part of that comment was just confusing. Kept it at 10 and removed the confusing part.

**Review by nickanderson — APPROVED** [review]

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/6377/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6377/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6377/

File: tests/acceptance/lib/reports/tail_n.cf.sub [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+      # Include the stdlib directly so that we are always testing the most
+      # recent version
+      inputs => { '../../../../lib/reports.cf' };
+}
+
+
+bundle agent main
+{
+  reports:
+    "My first line"
```
Comment (larsewi) [inline tests/acceptance/lib/reports/tail_n.cf.sub:12]: Should say my last lines

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/6399/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6399/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6399/

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/6454/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6454/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6454/

**Comment by nickanderson** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/6476/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6476/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6476/

**Comment by nickanderson** [comment]
The test needs to be skipped on versions prior to 3.18

**Comment by larsewi** [comment]
@cf-bottom Jenkins please !

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/6480/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6480/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6480/

**Review by nickanderson — APPROVED** [review]

**Comment by nickanderson** [comment]
The jenkins failure seems unrelated.

```
13:11:45  > git rev-parse refs/remotes/origin/pull/4572/merge^{commit} # timeout=10
13:11:46  > git rev-parse refs/remotes/origin/origin/pull/4572/merge^{commit} # timeout=10
13:11:46  > git rev-parse origin/pull/4572/merge^{commit} # timeout=10
13:11:46 ERROR: Couldn't find any revision to build. Verify the repository and branch configuration for this job.
13:11:46 Skipped archiving because build is not successful
13:11:46 Finished: FAILURE
```

**Comment by nickanderson** [comment]
@cf-bottom let's try jenkins, again, please.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/6485/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6485/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6485/

**Comment by nickanderson** [comment]
@cf-bottom let's try jenkins, again, please.

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/6543/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6543/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6543/

**Comment by larsewi** [comment]
@cf-bottom let's try jenkins, again, please.

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/6544/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/6544/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-6544/

**Pull request merged**